buku

```diff
@@ -4816,6 +4823,33 @@ def prompt(obj, results, noninteractive=False, deep=False, listtags=False, sugge
             show_taglist(bdb)
             continue
 
+        if (nav+' ').startswith('DB '):
+            dbpath, dbfile = os.path.split(bdb.dbfile)
+            if nav == 'DB':
+                print(f'Available DB files (in {dbpath}):')
+                for s in sorted(s for s in os.listdir(dbpath) if s.endswith('.db')):
+                    print(('*' if s == dbfile else ' '), s.removesuffix('.db'))
+            else:
+                s = os.path.expanduser(re.sub(r'^DB\s+', '', nav))
+                path = os.path.join(dbpath, (s if s != '~.' else BukuDb.get_default_dbdir()))  # relative to current dir
+                newpath, newfile = os.path.split(path+'/' if os.path.isdir(path) else path)
+                newfile, (_, ext) = newfile or 'bookmarks', os.path.splitext(newfile)
+                if not ext or os.path.isfile(os.path.join(newpath, newfile+'.db')):
+                    newfile += '.db'
+                newdb = os.path.join(newpath, newfile)
+                try:
+                    if not os.path.exists(newdb):
+                        print(f'DB file is being created at {newdb}')
+                    _bdb = bdb
+                    bdb = BukuDb(json=bdb.json, field_filter=bdb.field_filter, colorize=bdb.colorize, dbfile=newdb)
+                    _bdb.close()
+                    results, new_results = [], False
+                    cur_index = next_index = prev_index = 0
+                    print(f'Loaded DB file at {bdb.dbfile}')
+                except Exception:
+                    print(f'Failed to open DB file at {newdb}')
```

Comment:
Attempting to create a new `BukuDB` object for the specified file. If it succeeds, it replaces the old one (which is closed after that), and the search state gets reset. If it fails, the old `BukuDB` object is retained.

Appropriate messages are printed (not ending them with periods to avoid confusion; since they end on a filename).